app/src/main/java/com/github/geohunt/app/ui/components/profile/ProfilePage.kt

```diff
@@ -0,0 +1,103 @@
+package com.github.geohunt.app.ui.components.profile
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.NoLiveLiterals
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.geohunt.app.R
+import com.github.geohunt.app.model.LazyRef
+import com.github.geohunt.app.model.database.api.User
+import com.github.geohunt.app.model.database.firebase.FirebaseDatabase
+import com.github.geohunt.app.model.database.firebase.FirebaseUserRef
+import com.github.geohunt.app.ui.components.user.ProfileIcon
+import com.github.geohunt.app.ui.rememberLazyRef
+import com.github.geohunt.app.utility.findActivity
+
+@Composable
+fun ProfilePage(id: String) {
+    ProfilePage(user = FirebaseUserRef(id, FirebaseDatabase(LocalContext.current.findActivity())))
+}
+
+@Composable
+fun ProfilePage(user: LazyRef<User>) {
+    val lazyRefRemember = rememberLazyRef { user }
+
+    Box(modifier = Modifier
```

Comment:
You could use the FetchComponent Composable Guillaume made to avoid rewriting all this logic every time.